src/detection/DepthImageObstacleDetector.cc

```diff
@@ -284,25 +285,47 @@ int DepthImageObstacleDetector::extract_blobs()
 
                 blob_to_obstacle[label] = num_obstacles++;
                 obstacles[blob_to_obstacle[label]].id = label;
-                obstacles[blob_to_obstacle[label]].center.x = DBL_MAX;
+                obstacles[blob_to_obstacle[label]].bounding_box.tlc.y = i;
             }
 
             Obstacle *o = &obstacles[blob_to_obstacle[label]];
             o->center += glm::dvec3(0, i, j);
-            o->center.x = (depth_frame[row_offset + j] < o->center.x) ?
-                depth_frame[row_offset + j] : o->center.x;
+
+            if (depth_frame[row_offset + j] < o->bounding_box.tlc.x)
+                o->bounding_box.tlc.x = depth_frame[row_offset + j];
+
+            if (depth_frame[row_offset + j] > o->bounding_box.brf.x)
+                o->bounding_box.brf.x = depth_frame[row_offset + j];
+
+            if (i > o->bounding_box.brf.y)
+                o->bounding_box.brf.y = i;
+
+            if (j > o->bounding_box.brf.z)
+                o->bounding_box.brf.z = j;
+
+            if (j < o->bounding_box.tlc.z)
+                o->bounding_box.tlc.z = j;
         }
     }
 
     this->obstacles.resize(num_obstacles);
     for (Obstacle &o : obstacles) {
-        o.center.x *= this->scale;
+        o.bounding_box.brf.x *= this->scale;
+        o.bounding_box.tlc.x *= this->scale;
+
+        o.center.x = o.bounding_box.tlc.x;
```

Comment:
still, the radius shouldn't be the same. Even if the "center" is in the same plane as the front side of the bounding box, the radius on the corner should be bigger than the radius in the center of that plane, which will the the closest point of the object.